extmod/moducryptolib.c

```diff
@@ -234,19 +310,35 @@ STATIC mp_obj_t aes_process(size_t n_args, const mp_obj_t *args, bool encrypt) {
         }
     }
 
-    if (self->block_mode == UCRYPTOLIB_MODE_ECB) {
+    switch (self->block_mode) {
+    case UCRYPTOLIB_MODE_ECB: ;
         uint8_t *in = in_bufinfo.buf, *out = out_buf_ptr;
         uint8_t *top = in + in_bufinfo.len;
         for (; in < top; in += 16, out += 16) {
             aes_process_ecb_impl(&self->ctx, in, out, encrypt);
         }
-    } else {
+        break;
+
+    case UCRYPTOLIB_MODE_CBC:
         aes_process_cbc_impl(&self->ctx, in_bufinfo.buf, out_buf_ptr, in_bufinfo.len, encrypt);
+        break;
+
+#if MICROPY_PY_UCRYPTOLIB_CTR
+    case UCRYPTOLIB_MODE_CTR:
+        aes_process_ctr_impl(&self->ctx, in_bufinfo.buf, out_buf_ptr, in_bufinfo.len,
+            ctr_params_from_aes(self));
+#endif
+        break;
+
+    default:
+        MP_UNREACHABLE();
+        break;
     }
 
     if (out_buf != MP_OBJ_NULL) {
         return out_buf;
     }
+
```

Comment:
Unrelated change.

Reply:
It's indeed unrelated to the commit in which I've added it. But since that commit was a bit of an overhaul of that function, I think it's okay to push into it a few "random" improvements (I think we do agree that there should be a blank there, and if I see "bad" code in an area which I'm about to modify anyway, I'm willing to do so). Question is whether it should be placed in a separate commit or not.

Reply:
Okay, found https://github.com/micropython/micropython/pull/4727#discussion_r279917327. I'll move to a separate comit.

Reply:
> I think we do agree that there should be a blank there

No, I personally don't agree. MicroPython codestyle is more "tight" than other codestyles out there. Extra empty lines are unlikely needed. And it's definitely not worth to introduce commit noise which doesn't affect functionality but complicate change review. The above should give another reason why - if there's anything to agree on, it's that the random changes which don't bring any real value are to be avoided. If that's bypassed, there unlikely will be any further agreement, and what's left is different guys erasing and flip-flopping each other's changes in infinite loop.

Reply:
> what's left is different guys erasing and flip-flopping each other's changes in infinite loop.

Well, I can definitely see that happening.

Okay, I'll remove that formatting commit.